tests/test_sensor.py

```diff
@@ -262,3 +307,72 @@ def test_remove_callbacks(self, ss, sensors, add_callback, remove_callback):
         ss.remove(sensors[0])
         remove_callback.assert_not_called()
         add_callback.assert_not_called()
+
+
+class MyAgg(AggregateSensor):
+    """A simple AggregateSensor subclass for testing."""
+
+    def update_aggregate(
+        self,
+        updated_sensor: Optional[Sensor],
+        reading: Optional[Reading],
+        old_reading: Optional[Reading],
+    ) -> Reading:
+        """Return a known Reading."""
+        return Reading(0, Sensor.Status.NOMINAL, 7)
+
+
+@pytest.fixture
+def agg_sensor(mocker, ss):
+    """Mock out update_aggregate so we can check it's called appropriately."""
+
+    def return_a_reading(*args, **kwargs):
+        """Return an arbitrary Reading."""
+        return Reading(0, Sensor.Status.WARN, 8)
+
+    my_agg = MyAgg(target=ss, sensor_type=int, name="good-bad-ugly")
+    mocker.patch.object(my_agg, "update_aggregate", side_effect=return_a_reading)
+    return my_agg
+
+
+class TestAggregateSensor:
+    """Test operation of AggregateSensor."""
+
+    def test_creation(self, agg_sensor, ss, sensors):
+        """Check that creation happens properly, and correct initial value is set."""
+        assert agg_sensor.target is ss
+        assert agg_sensor.reading.timestamp == 0
+        assert agg_sensor.reading.status == Sensor.Status.NOMINAL
+        assert agg_sensor.reading.value == 7
```

Comment:
This confused me a bit, because `return_a_reading` returns 8, not 7. I realised it's because you're mocking the object after it gets created, and so the constructor calls the original version. I feel like it would be more consistent to mock the class so that all uses of `update_aggregate` get intercepted.

Reply:
I tried to get the mock working before the object is created, but I couldn't actually accomplish it in the end. I've tried for the last little while, but I can't figure it out. Do you have an example somewhere that mocks out an entire class?